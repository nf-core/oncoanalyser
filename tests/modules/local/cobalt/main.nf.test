nextflow_process {

    name "Test Process COBALT"
    script "modules/local/cobalt/main.nf"
    process "COBALT"

    test("COBALT testing tumor + normal") {

        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/markdups_bam/'// define parameters here. Example:
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")

                input[0] = [
                    [id: 'test',
                    tumor_id: 'tumor_test',
                    normal_id: 'normal_test'
                    ],
                
                    file(params.test_sample_data +  'subject_a.tumor.dna.bwa-mem2_2.2.1.markdups.bam', checkIfExists: true),
                    file(params.test_sample_data +  'subject_a.normal.dna.bwa-mem2_2.2.1.markdups.bam', checkIfExists: true),
                    file(params.test_sample_data +  'subject_a.tumor.dna.bwa-mem2_2.2.1.markdups.bam.bai', checkIfExists: true),
                    file(params.test_sample_data +  'subject_a.normal.dna.bwa-mem2_2.2.1.markdups.bam.bai', checkIfExists: true),
                
                ]
                input[1] = file("/home/rhassaine/resource_files_oa/38/results/reference_data/1.0.0/20250203_101759/hmf_data_5.34_38--2/dna_pipeline/copy_number/GC_profile.1000bp.38.cnp")
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            def cobalt_output_dir = process.out.cobalt_dir[0][1]
            
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert path(cobalt_output_dir).resolve("cobalt.version").exists() },
                { assert path(cobalt_output_dir).resolve("tumor_test.cobalt.gc.median.tsv").exists() },
                { assert path(cobalt_output_dir).resolve("tumor_test.cobalt.ratio.pcf").exists() },
                { assert path(cobalt_output_dir).resolve("tumor_test.cobalt.ratio.tsv.gz").exists() },
                { assert path(cobalt_output_dir).resolve("normal_test.cobalt.gc.median.tsv").exists() },
                { assert path(cobalt_output_dir).resolve("normal_test.cobalt.ratio.median.tsv").exists() },
                { assert path(cobalt_output_dir).resolve("normal_test.cobalt.ratio.pcf").exists() },
                { assert path(process.out.versions[0]).exists() }
            )
        }

    }

    test("COBALT with tumor only") {
        when {
            params {
                // publish_dir_mode = "symlink"
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/markdups_bam/'
            }
            process {
                """
                input[0] = [
                    [id: 'test', tumor_id: 'tumor_test'],
                    file(params.test_sample_data + 'subject_a.tumor.dna.bwa-mem2_2.2.1.markdups.bam', checkIfExists: true),
                    [],
                    file(params.test_sample_data + 'subject_a.tumor.dna.bwa-mem2_2.2.1.markdups.bam.bai', checkIfExists: true),
                    []
                ]
                input[1] = file("/home/rhassaine/resource_files_oa/38/results/reference_data/1.0.0/20250203_101759/hmf_data_5.34_38--2/dna_pipeline/copy_number/GC_profile.1000bp.38.cnp")
                input[2] = file("/home/rhassaine/resource_files_oa/38/results/reference_data/1.0.0/20250203_101759/hmf_data_5.34_38--2/dna_pipeline/copy_number/DiploidRegions.38.bed.gz")
                input[3] = []
                """
            }
        }

       then {
        def cobalt_output_dir = process.out.cobalt_dir[0][1]
        
        assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() },
            { assert path(cobalt_output_dir).resolve("cobalt.version").exists() },
            { assert path(cobalt_output_dir).resolve("tumor_test.cobalt.gc.median.tsv").exists() },
            { assert path(cobalt_output_dir).resolve("tumor_test.cobalt.ratio.pcf").exists() },
            { assert path(cobalt_output_dir).resolve("tumor_test.cobalt.ratio.tsv.gz").exists() },
            { assert path(process.out.versions[0]).exists() }
    )
}
        }
    }
