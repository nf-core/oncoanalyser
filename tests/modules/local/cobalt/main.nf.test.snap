{
    "COBALT with tumor only": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "tumor_id": "tumor_test"
                        },
                        [
                            "cobalt.version:md5,ef44c6894423adb9fe3c15f714ec947b",
                            "tumor_test.cobalt.gc.median.tsv:md5,c80a997fd0dc3226aefae4e4e11926d5",
                            "tumor_test.cobalt.ratio.pcf:md5,dd979e7f86c68ff02715ffdc12e8557d",
                            "tumor_test.cobalt.ratio.tsv.gz:md5,dc8cb6e711b60ae7e0dfee76cd49c7ca"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,7f3996abc1b9e36d57d5ebf289e7e2ec"
                ],
                "cobalt_dir": [
                    [
                        {
                            "id": "test",
                            "tumor_id": "tumor_test"
                        },
                        [
                            "cobalt.version:md5,ef44c6894423adb9fe3c15f714ec947b",
                            "tumor_test.cobalt.gc.median.tsv:md5,c80a997fd0dc3226aefae4e4e11926d5",
                            "tumor_test.cobalt.ratio.pcf:md5,dd979e7f86c68ff02715ffdc12e8557d",
                            "tumor_test.cobalt.ratio.tsv.gz:md5,dc8cb6e711b60ae7e0dfee76cd49c7ca"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7f3996abc1b9e36d57d5ebf289e7e2ec"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-03T13:02:48.873024669"
    },
    "COBALT testing tumor + normal": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "tumor_id": "tumor_test",
                            "normal_id": "normal_test"
                        },
                        [
                            "cobalt.version:md5,ef44c6894423adb9fe3c15f714ec947b",
                            "normal_test.cobalt.gc.median.tsv:md5,9deaecdc8199d76bb3114ca3167e2571",
                            "normal_test.cobalt.ratio.median.tsv:md5,b361c09c0bea5e42156a638316afdfe2",
                            "normal_test.cobalt.ratio.pcf:md5,142be492e8031846f8e2235661b8525f",
                            "tumor_test.cobalt.gc.median.tsv:md5,c80a997fd0dc3226aefae4e4e11926d5",
                            "tumor_test.cobalt.ratio.pcf:md5,b713be54407649e73654c8a9501c3951",
                            "tumor_test.cobalt.ratio.tsv.gz:md5,5834b95681366765f023c0aa9b142b71"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,7f3996abc1b9e36d57d5ebf289e7e2ec"
                ],
                "cobalt_dir": [
                    [
                        {
                            "id": "test",
                            "tumor_id": "tumor_test",
                            "normal_id": "normal_test"
                        },
                        [
                            "cobalt.version:md5,ef44c6894423adb9fe3c15f714ec947b",
                            "normal_test.cobalt.gc.median.tsv:md5,9deaecdc8199d76bb3114ca3167e2571",
                            "normal_test.cobalt.ratio.median.tsv:md5,b361c09c0bea5e42156a638316afdfe2",
                            "normal_test.cobalt.ratio.pcf:md5,142be492e8031846f8e2235661b8525f",
                            "tumor_test.cobalt.gc.median.tsv:md5,c80a997fd0dc3226aefae4e4e11926d5",
                            "tumor_test.cobalt.ratio.pcf:md5,b713be54407649e73654c8a9501c3951",
                            "tumor_test.cobalt.ratio.tsv.gz:md5,5834b95681366765f023c0aa9b142b71"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7f3996abc1b9e36d57d5ebf289e7e2ec"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-03T13:00:59.296181532"
    }
}